[None][fix] Use class-qualified access for clarity

Fan-Yunfan opened the pull request:
### Problem
`cpp/include/tensorrt_llm/runtime/iTensor.h`

1.  In the `getDimension()` method, the `static_assert` used for compile-time checking utilizes the static constant attribute `MAX_DIMS` of the runtime object "`shape`", which may cause confusion for readers: why can "`shape`", clearly a runtime object, be used at compile time?
```
    using Shape = nvinfer1::Dims;
    ......
    [[nodiscard]] virtual Shape const& getShape() const = 0;
    ......
    template <SizeType32 n>
    [[nodiscard]] DimType64 getDimension() const
    {
        auto const shape = getShape();
        static_assert(n < shape.MAX_DIMS && n >= -shape.MAX_DIMS,
            "Trying to access the dimension of a tensor, when its maximal shape cannot have that dimension.");
        if constexpr (n < 0)
        {
            return shape.d[shape.nbDims + n];
        }
        else
        {
            return shape.d[n];
        }
    }
```
`TensorRT/include/NvInferRuntimeBase.h
`
```
//!
//! Alias for Dims64.
//!
using Dims = Dims64;
class Dims64
{
public:
    //! The maximum rank (number of dimensions) supported for a tensor.
    static constexpr int32_t MAX_DIMS{8};

    //! The rank (number of dimensions).
    int32_t nbDims;

    //! The extent of each dimension.
    int64_t d[MAX_DIMS];
};
```
2. In the `volumeNonNegative() `method, the data type of "`vol`" is `std::int64_t`, but it is converted to the `std::size_t` type when returned. On 32-bit platforms, converting `std::int64_t` to `std::size_t` may pose an overflow risk.
```
static std::int64_t volume(Shape const& dims)
{
    {
        return dims.nbDims < 0 ? -1
            : dims.nbDims == 0
            ? 0
            : std::accumulate(dims.d, dims.d + dims.nbDims, std::int64_t{1}, std::multiplies<>{});
    }
}
......
static std::size_t volumeNonNegative(Shape const& shape)
{
    auto const vol = volume(shape);
    TLLM_CHECK_WITH_INFO(0 <= vol, "Invalid tensor shape");
    return static_cast<std::size_t>(vol);
}
```
### Solution
1. Replace static member access through object instances (which appears to be runtime but is optimized to compile-time by the compiler) with class type scope access (compile-time)
```
template <SizeType32 n>
[[nodiscard]] DimType64 getDimension() const
{
    auto const shape = getShape();
    static_assert(n < Shape::MAX_DIMS && n >= -Shape::MAX_DIMS,
        "Trying to access the dimension of a tensor, when its maximal shape cannot have that dimension.");
    ......
}
```
2. Add boundary checks for 32-bit platforms
```
if constexpr (sizeof(std::size_t) == 4)
{
    TLLM_CHECK_WITH_INFO(vol <= static_cast<std::int64_t>(std::numeric_limits<std::size_t>::max(),
                             "Tensor volume exceeds 32-bit size_t maximum capacity."));
}
```






<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fix**
  * Updated `ITensor::getDimension` to use `Shape::MAX_DIMS` in the compile-time assertion.

## Dev Engineer Review

* The change is limited to `cpp/include/tensorrt_llm/runtime/iTensor.h`.
* The explicit `Shape::MAX_DIMS` reference improves compile-time clarity and API consistency.
* No `volumeNonNegative` boundary check is present in the current diff.
* No configuration, public declaration, or test-list files changed.
* No performance or regression concerns are evident from this one-line change.

## QA Engineer Review

* No test changes.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
[FIX] fix bugs caused by None attention_bias during qwen3 model conversion

Signed-off-by: fanyunfan <2569548856@qq.com>

Fan-Yunfan committed:
Merge branch 'main' into main

Fan-Yunfan committed:
Merge branch 'NVIDIA:main' into main

Fan-Yunfan committed:
Merge branch 'NVIDIA:main' into main

Fan-Yunfan committed:
Merge branch 'NVIDIA:main' into main

Fan-Yunfan committed:
[None][fix] Enhancing code robustness and adding boundary checks for ITensor

Signed-off-by: fanyunfan <2569548856@qq.com>

Fan-Yunfan commented:
Dear @karljang, I've made some code quality improvements for ITensor. If the review looks good, we can squash the commits from this PR and https://github.com/NVIDIA/TensorRT-LLM/pull/8140 into a single commit to reduce the validation pressure on our CI pipeline.
![9a3430c5-d2d0-4316-9128-47c2cc007efc](https://github.com/user-attachments/assets/8d99d502-72da-4b78-aa1e-45e03dd4704b)

tongyuantongyu commented on cpp/include/tensorrt_llm/runtime/iTensor.h:
```diff
@@ -120,6 +120,11 @@ class ITensor : virtual public IBuffer
     {
         auto const vol = volume(shape);
         TLLM_CHECK_WITH_INFO(0 <= vol, "Invalid tensor shape");
+        if constexpr (sizeof(std::size_t) == 4)
```
We don't support 32bit platform. The only supported platforms are `x86_64-linux-gnu` and `aarch64-linux-gnu`.

Fan-Yunfan replied:
Got it, thanks!

tongyuantongyu commented on cpp/include/tensorrt_llm/runtime/iTensor.h:
```diff
@@ -65,7 +65,7 @@ class ITensor : virtual public IBuffer
     {
         auto const shape = getShape();
-        static_assert(n < shape.MAX_DIMS && n >= -shape.MAX_DIMS,
+        static_assert(n < Shape::MAX_DIMS && n >= -Shape::MAX_DIMS,
```
Access `static constepxr` from object expression is valid standard C++ syntax. It's indeed better to use the class scope, but this change is better described as "use class-qualified access for clarity".

Fan-Yunfan replied:
Indeed — thanks for the review.

tongyuantongyu commented:
/bot run

tongyuantongyu commented:
🚧Please don't attempt rebase or merge main for this PR unless otherwise asked from now on.

tensorrt-cicd commented:
[PR_Github #66606](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/66606/) [ run ] triggered by Bot. Commit: `c2c1c11` [Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5311137534)

tensorrt-cicd commented:
[PR_Github #66606](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/66606/) [ run ] completed with state `SUCCESS`. Commit: `c2c1c11` 
[/LLM/main/L0_MergeRequest_PR pipeline #54230](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/54230/) completed with status: 'FAILURE'

[CI Report](http://tensorrt-llm.tensorrt-llm-ci-report.sc2-paas.nvidia.com/?job=%2FLLM%2Fmain%2FL0_MergeRequest_PR&build=54230)

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again


[Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5311137534)

Fan-Yunfan commented:
> 🚧Please don't attempt rebase or merge main for this PR unless otherwise asked from now on.

Got it～

tongyuantongyu commented:
/bot run

tensorrt-cicd commented:
[PR_Github #67317](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/67317/) [ run ] triggered by Bot. Commit: `c2c1c11` [Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5338310618)

tensorrt-cicd commented:
[PR_Github #67317](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/67317/) [ run ] completed with state `FAILURE`. Commit: `c2c1c11` 
[/LLM/main/L0_MergeRequest_PR pipeline #54841](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/54841/) completed with status: 'FAILURE'

[CI Report](http://tensorrt-llm.tensorrt-llm-ci-report.sc2-paas.nvidia.com/?job=%2FLLM%2Fmain%2FL0_MergeRequest_PR&build=54841)

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again


[Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5338310618)

tongyuantongyu commented:
/bot run

tensorrt-cicd commented:
[PR_Github #67348](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/67348/) [ run ] triggered by Bot. Commit: `e290418` [Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5339016199)

tensorrt-cicd commented:
[PR_Github #67348](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/67348/) [ run ] completed with state `FAILURE`. Commit: `e290418` 


[Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5339016199)

tongyuantongyu commented:
@Fan-Yunfan could you squash your commits? Your branch 's base is too old.

Fan-Yunfan commented:
> @Fan-Yunfan could you squash your commits? Your branch 's base is too old.

Thanks for you reminder, I have rebased the latest main branch and squashed serval commits to  one commit just now.

tongyuantongyu commented:
/bot run

tensorrt-cicd commented:
[PR_Github #67657](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/67657/) [ run ] triggered by Bot. Commit: `256c01a` [Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5351362433)

tensorrt-cicd commented:
[PR_Github #67657](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/67657/) [ run ] completed with state `FAILURE`. Commit: `256c01a` 
[/LLM/main/L0_MergeRequest_PR pipeline #55144](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/55144/) completed with status: 'FAILURE'

[CI Report](http://tensorrt-llm.tensorrt-llm-ci-report.sc2-paas.nvidia.com/?job=%2FLLM%2Fmain%2FL0_MergeRequest_PR&build=55144)

⚠️ **Action Required:**
- Please check the failed tests and fix your PR
- If you cannot view the failures, ask the CI triggerer to share details
- Once fixed, request an NVIDIA team member to trigger CI again

[CI Agent Failure Analysis](https://pbss.s8k.io/v1/AUTH_svc_tensorrt/sw-tensorrt-ci-analysis/LLM/main/L0_MergeRequest_PR/55144/failure_analysis.html)


[Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5351362433)

tongyuantongyu commented:
/bot skip --comment "Known infra issue, build passed"

tensorrt-cicd commented:
[PR_Github #68105](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/68105/) [ skip ] triggered by Bot. Commit: `256c01a` [Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5364643149)

tensorrt-cicd commented:
[PR_Github #68105](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/68105/) [ skip ] completed with state `SUCCESS`. Commit: `256c01a` 
Skipping testing for commit `256c01a`


[Link to invocation](https://github.com/NVIDIA/TensorRT-LLM/pull/8855#issuecomment-5364643149)
